tests/dispatch/tests.py

```diff
@@ -126,8 +149,28 @@ def uid_based_receiver_2(**kwargs):
         a_signal.disconnect(dispatch_uid="uid")
         self.assertTestIsClean(a_signal)
 
-    def test_robust(self):
-        """Test the send_robust() function"""
+    def test_send_robust_success(self):
+        a_signal.connect(receiver_1_arg)
+        result = a_signal.send_robust(sender=self, val="test")
+        expected = [(receiver_1_arg, "test")]
+        self.assertEqual(result, expected)
+        a_signal.disconnect(receiver_1_arg)
+        self.assertTestIsClean(a_signal)
+
+    def test_send_robust_no_receivers(self):
+        result = a_signal.send_robust(sender=self, val="test")
+        expected = []
+        self.assertEqual(result, expected)
+
+    def test_send_robust_ignored_sender(self):
+        a_signal.connect(receiver_1_arg)
+        expected = [(receiver_1_arg, "test")]
+        result = a_signal.send_robust(sender=self, val="test")
+        self.assertEqual(result, expected)
+        a_signal.disconnect(receiver_1_arg)
```

Comment:
If the assertion above fails, they signal won't get disconnected. I think it could affect test isolation, correct? I guess some existing tests have that issue also. Maybe a future enhancement could be a context manager to allow connect/disconnect without having to write try/finally everywhere? I'm not insistenting on using try/finally now.